import-automation/workflow/spanner-ingestion-workflow.yaml

```diff
@@ -0,0 +1,156 @@
+main:
+  params: [args]
+  steps:
+    - invoke_ingestion_workflow:
+        try:
+          #call: googleapis.workflowexecutions.v1.projects.locations.workflows.executions.run
+          call: spanner_ingestion_workflow
+          args:
+            workflow_id: 'spanner-ingestion-workflow' 
+          result: workflow_result
+        except:
+          as: e
+          steps:
+            - clean_up:
+                call: http.post
+                args:
+                  url: 'https://spanner-ingestion-helper-965988403328.us-central1.run.app'
+                  auth:
+                    type: OIDC
+                  body:
+                    actionType: release_ingestion_lock
+                    workflowId: '${sys.get_env("GOOGLE_CLOUD_WORKFLOW_EXECUTION_ID")}'
+                result: function_response
+            - fail_workflow:
+                raise: ${e}
+    - return_result:
+        return: ${workflow_result}
+
+# This sub-workflow performs the following steps
+# - Acquire ingestion lock
+# - Get the list of pending imports
+# - Run dataflow ingestion job
+# - Update import status in spanner
+# - Release ingestion lock
+spanner_ingestion_workflow:
+  params: [workflow_id]
+  steps:
+    - init:
+        assign:
+          - lock_timeout: 3600 #seconds
```

Comment:
Just to double check, what will happen if the lock times out before a job completes? Is 1 hr sufficient for larger imports?